src/dune_lang/syntax.mli

```diff
@@ -72,8 +72,10 @@ val name : t -> string
 (** Check that the given version is supported and raise otherwise. *)
 val check_supported : dune_lang_ver:Version.t -> t -> Loc.t * Version.t -> unit
 
-val greatest_supported_version :
-  ?dune_lang_ver:Version.t -> t -> Version.t option
+val greatest_supported_version : t -> Version.t
```

Comment:
@voodoos seems like it's best to separate the cases where `dune_lang_ver` is passed. There should always be a greatest supported version if we omit this argument.

Reply:
Yes, I agree with you, it reads better that way.